internal/querynode/data_sync_service.go

```diff
@@ -42,11 +42,42 @@ type dataSyncService struct {
 	msFactory         msgstream.Factory
 }
 
+// checkReplica used to check replica info before init flow graph, it's a private method of dataSyncService
+func (dsService *dataSyncService) checkReplica(collectionID UniqueID) error {
+	// check if the collection exists
+	hisColl, err := dsService.historicalReplica.getCollectionByID(collectionID)
```

Comment:
The way to get collection from both streamingReplica and historical replica looks weird

Reply:
I'll merge these, see https://github.com/milvus-io/milvus/issues/17160